nuclio_sdk/context.py

```diff
@@ -28,3 +43,16 @@ def __init__(self, logger=None, platform=None, worker_id=None, trigger=None):
         # for backwards compatibility
         if self.trigger is not None:
             self.trigger_name = self.trigger.name
+
+    @property
+    def logger(self):
+        if self._logger_per_async_tasks:
+            return self._logger.get() or self._shared_logger
+        return self._logger
+
+    @logger.setter
+    def logger(self, logger):
+        if self._logger_per_async_tasks:
+            self._logger.set(logger)
+        else:
+            self._logger = logger
```

Comment:
Just so I understand - the plan is to create a new logger for each task and set it on the context?
The issue is that the context is shared (in nuclio's common wrapper) so you will replace it for other tasks.